elliptic-curve/src/hash2field/expand_msg.rs

```diff
@@ -65,10 +72,11 @@ impl<L: ArrayLength<u8>> Domain<L> {
         }
     }
 
-    pub fn len(&self) -> usize {
+    pub fn len(&self) -> u8 {
         match self {
-            Self::Hashed(_) => L::to_usize(),
-            Self::Array(d) => d.len(),
+            Self::Hashed(_) => L::to_u8(),
+            // Can't overflow because enforced on a type level.
+            Self::Array(d) => d.len() as u8,
```

Comment:
You changed the bound to `IsLessOrEqual<U256>`, but `256usize as u8` overflows to 0?

Are you preventing that overflow in some way I'm missing?